File: C++/题目1167：数组排序.cpp

// 题目1167：数组排序

/**
 * @author:wangzq
 * @email:wangzhenqing1008@163.com
 * @date:2015-06-30 11:01:53
 * @url:http://ac.jobdu.com/problem.php?pid=1167
 * 解题思路参考csdn:http://blog.csdn.net/u013027996/article/details/17224793
 */

#include <stdio.h>
#include <algorithm>
#include <map>

using namespace std;
const int maxn = 10002;
int i, n;
int array[maxn];
int newArray[maxn];
map<int, int> numMap;

int main() {
    while (scanf("%d", &n) != EOF) {
        for (i = 0; i < n; i++) {
            scanf("%d", &array[i]);
            newArray[i] = array[i];
        }
        sort(newArray, newArray + n);
        int index = 1;
        map<int, int>::iterator it;
        for (i = 0; i < n; i++) {
            it = numMap.find(newArray[i]);
            if (it == numMap.end()) {
                numMap.insert(make_pair(newArray[i], index));
                index++;
            }
        }
        for (i = 0; i < n - 1; i++) {
            printf("%d ", numMap.find(array[i])->second);
        }
        printf("%d\n", numMap.find(array[n - 1])->second);
    }
    return 0;
}
/**************************************************************
Problem: 1167
User: wangzhenqing
Language: C++
Result: Accepted
Time:20 ms
Memory:1504 kb
****************************************************************/


                        

File: Java/题目1167：数组排序.java

// 题目1167：数组排序

/**
 * @author:wangzq
 * @email:wangzhenqing1008@163.com
 * @date:2015-06-30 11:01:53
 * @url:http://ac.jobdu.com/problem.php?pid=1167 解题思路参考csdn:http://blog.csdn.net/u013027996/article/details/17224793
 */

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

/**
 * @ClassName: Main1046
 * @Description: TODO
 * @author wangzq
 * @date 2013-8-7 下午04:00:39
 *
 * @version 3.0.0
 */
public class Main {
    /*
     * 1167
     */
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        while (scanner.hasNext()) {
            int n = scanner.nextInt();
            Map<Integer, Integer> nuMap = new HashMap<Integer, Integer>();
            int array[] = new int[n];
            int newArray[] = new int[n];
            for (int i = 0; i < n; i++) {
                array[i] = scanner.nextInt();
                newArray[i] = array[i];
            }
            Arrays.sort(newArray);
            int index = 1;
            for (int i = 0; i < n; i++) {
                if (!nuMap.containsKey(newArray[i])) {
                    nuMap.put(newArray[i], index);
                    index++;
                }
            }

            for (int i = 0; i < n - 1; i++) {
                System.out.print(nuMap.get(array[i]) + " ");
            }
            System.out.print(nuMap.get(array[n - 1]));
            System.out.println();
        }
    }
}
/**************************************************************
 Problem: 1167
 User: wzqwsrf
 Language: Java
 Result: Accepted
 Time:740 ms
 Memory:37612 kb
 ****************************************************************/




